### DISCUSSION TITLE
Update Profiles Table URL When Avatar is Added to Storage

### DISCUSSION THREAD

**Discussion by boilermanc** [supabase/supabase#20789] [Questions]
Been struggling with this for a couple of days. When a user uploads a new avatar it goes into a bucket and folder. I created a function that will generate a url when it is triggered by the new upload in the objects table. that works. but of course the url is not correct and needs a token. cannot really figure that out or if there is a better way to do this. pretty common stuff...

**Comment by GaryAustin1** [comment]
What sort of token are you referring to?
Normally avatars are in a public bucket so the url is fixed based on the path and some constants.
Are you doing a private bucket?  If so normally you would store the path in the db.  And then when you want to use it call createSignedUrl() to get a complete url as the url will expire.

Also note that upload currently involves two inserts and you will have your function called twice.
